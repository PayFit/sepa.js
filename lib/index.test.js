const SEPA = require('.');

describe('#SEPA', () => {
  it('should generate a full SEPA', () => {
    const doc = new SEPA.Document('pain.008.001.02');
    doc.grpHdr.id = '1️⃣XMPL.20140201.TR0';
    doc.grpHdr.created = new Date('2019-01-01T08:00');
    doc.grpHdr.initiatorName = '1️⃣Example & LLC@';

    const info = doc.createPaymentInfo();
    info.collectionDate = new Date('2019-01-01T08:00');
    info.creditorIBAN = 'DE87123456781234567890';
    info.creditorBIC = 'XMPLDEM0XXX';
    info.creditorName =
      'ASSOCIATION GROUPEMENT POUR LA DIRECTION ET LA GESTION DES CENTRES DE SOINS SHOULD BE CUT';
    info.creditorId = 'DE98ZZZ09999999999';
    info.batchBooking = true;
    doc.addPaymentInfo(info);

    const tx = info.createTransaction();
    tx.debtorName = "Jean L'Och";
    tx.debtorIBAN = 'DE40987654329876543210';
    tx.debtorBIC = 'CUSTDEM0XXX';
    tx.mandateId = '_X&M@PL.CUST487.2014';
    tx.mandateSignatureDate = new Date('2014-02-01');
    tx.amount = 50.23;
    tx.currency = 'EUR';
    tx.remittanceInfo = 'INVOICE 54';
    tx.end2endId = '1️⃣X&M@.CUST487.INVOICE.54';
    info.addTransaction(tx);

    expect(doc.toString()).toMatchSnapshot();

    expect(tx.end2endId).toBe('1XM.CUST487.INVOICE.54');
    expect(tx.debtorName).toBe('Jean LOch');
    expect(tx.mandateId).toBe('XMPL.CUST487.2014');

    expect(doc.grpHdr.initiatorName).toBe('1Example  LLC');
    expect(doc.grpHdr.id).toBe('1XMPL.20140201.TR0');

    expect(info.creditorName).toBe(
      'ASSOCIATION GROUPEMENT POUR LA DIRECTION ET LA GESTION DES CENTRES DE '
    );
  });
});
