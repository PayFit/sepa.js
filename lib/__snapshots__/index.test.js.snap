// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#SEPA should generate a full SEPA 1`] = `"<?xml version="1.0" encoding="UTF-8"?><Document xmlns="urn:iso:std:iso:20022:tech:xsd:pain.008.001.02" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:iso:std:iso:20022:tech:xsd:pain.008.001.02 pain.008.001.02.xsd"><CstmrDrctDbtInitn><GrpHdr><MsgId>1XMPL.20140201.TR0</MsgId><CreDtTm>2019-01-01T07:00:00.000Z</CreDtTm><NbOfTxs>1</NbOfTxs><CtrlSum>50.23</CtrlSum><InitgPty><Nm>1Example  LLC</Nm></InitgPty></GrpHdr><PmtInf><PmtInfId>1XMPL.20140201.TR0.0</PmtInfId><PmtMtd>DD</PmtMtd><BtchBookg>true</BtchBookg><NbOfTxs>1</NbOfTxs><CtrlSum>50.23</CtrlSum><PmtTpInf><SvcLvl><Cd>SEPA</Cd></SvcLvl><LclInstrm><Cd>CORE</Cd></LclInstrm><SeqTp>FRST</SeqTp></PmtTpInf><ReqdColltnDt>2019-01-01</ReqdColltnDt><Cdtr><Nm>ASSOCIATION GROUPEMENT POUR LA DIRECTION ET LA GESTION DES CENTRES DE </Nm></Cdtr><CdtrAcct><Id><IBAN>DE87123456781234567890</IBAN></Id></CdtrAcct><CdtrAgt><FinInstnId><BIC>XMPLDEM0XXX</BIC></FinInstnId></CdtrAgt><ChrgBr>SLEV</ChrgBr><CdtrSchmeId><Id><PrvtId><Othr><Id>DE98ZZZ09999999999</Id><SchmeNm><Prtry>SEPA</Prtry></SchmeNm></Othr></PrvtId></Id></CdtrSchmeId><DrctDbtTxInf><PmtId><InstrId>1XMPL.20140201.TR0.0.0</InstrId><EndToEndId>1XM.CUST487.INVOICE.54</EndToEndId></PmtId><InstdAmt Ccy="EUR">50.23</InstdAmt><DrctDbtTx><MndtRltdInf><MndtId>XMPL.CUST487.2014</MndtId><DtOfSgntr>2014-02-01</DtOfSgntr><AmdmntInd>false</AmdmntInd></MndtRltdInf></DrctDbtTx><DbtrAgt><FinInstnId><BIC>CUSTDEM0XXX</BIC></FinInstnId></DbtrAgt><Dbtr><Nm>Jean LOch</Nm></Dbtr><DbtrAcct><Id><IBAN>DE40987654329876543210</IBAN></Id></DbtrAcct><RmtInf><Ustrd>INVOICE 54</Ustrd></RmtInf></DrctDbtTxInf></PmtInf></CstmrDrctDbtInitn></Document>"`;
